components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1251,30 +1251,66 @@ where
                 // already is a force leader, do nothing
                 return;
             }
+            Some(ForceLeaderState::WaitTicks { .. }) => {
+                self.fsm.peer.force_leader = None;
+            }
             None => {}
         }
 
-        if self.fsm.peer.is_leader() {
-            warn!(
-                "reject pre force leader due to already being leader";
-                "region_id" => self.fsm.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-            );
-            return;
-        }
-
+        let ticks = if self.fsm.peer.is_leader() {
+            if self.fsm.hibernate_state.group_state() == GroupState::Ordered {
+                warn!(
+                    "reject pre force leader due to already being leader";
+                    "region_id" => self.fsm.region_id(),
+                    "peer_id" => self.fsm.peer_id(),
+                );
+                return;
+            }
+            // wait two rounds of election timeout to trigger check quorum to step down the leader
+            Some(
+                (self.ctx.cfg.raft_election_timeout_ticks * 2)
+                    .saturating_sub(self.fsm.peer.raft_group.raft.election_elapsed)
+                    + 1,
+            )
         // When election timeout is triggered, leader_id is set to INVALID_ID.
         // But learner(not promotable) is a exception here as it wouldn't tick
         // election.
-        if self.fsm.peer.raft_group.raft.promotable()
-            && self.fsm.hibernate_state.group_state() != GroupState::Idle
+        } else if self.fsm.peer.raft_group.raft.promotable()
             && self.fsm.peer.leader_id() != raft::INVALID_ID
         {
-            warn!(
-                "reject pre force leader due to leader lease may not expired";
+            if self.fsm.hibernate_state.group_state() == GroupState::Ordered {
+                warn!(
+                    "reject pre force leader due to leader lease may not expired";
+                    "region_id" => self.fsm.region_id(),
+                    "peer_id" => self.fsm.peer_id(),
+                );
+                return;
+            }
+            // wait one round of election timeout to make sure leader_id is invalid
```

Comment:
Leader is always valid until it starts election.

Reply:
Yes, so wait timeout to trigger election

Reply:
What's your point here?

Reply:
elapsed passes election timeout doesn't mean it will start elections. Should check with double timeout.

Reply:
oh, should be `randomized_election_timeout`